ext/nnstreamer/tensor_decoder/tensordec-pose.c

```diff
@@ -0,0 +1,522 @@
+    draw_line_with_dot (frame, data, XYdata[1]->x, XYdata[1]->y, XYdata[8]->x,
+        XYdata[8]->y);
+  if (XYdata[8]->valid && XYdata[9]->valid)
+    draw_line_with_dot (frame, data, XYdata[8]->x, XYdata[8]->y, XYdata[9]->x,
+        XYdata[9]->y);
+  if (XYdata[9]->valid && XYdata[10]->valid)
+    draw_line_with_dot (frame, data, XYdata[9]->x, XYdata[9]->y,
+        XYdata[10]->x, XYdata[10]->y);
+  if (XYdata[1]->valid && XYdata[11]->valid)
+    draw_line_with_dot (frame, data, XYdata[1]->x, XYdata[1]->y,
+        XYdata[11]->x, XYdata[11]->y);
+  if (XYdata[11]->valid && XYdata[12]->valid)
+    draw_line_with_dot (frame, data, XYdata[11]->x, XYdata[11]->y,
+        XYdata[12]->x, XYdata[12]->y);
+  if (XYdata[12]->valid && XYdata[13]->valid)
+    draw_line_with_dot (frame, data, XYdata[12]->x, XYdata[12]->y,
+        XYdata[13]->x, XYdata[13]->y);
+  draw_label (frame, data, *XYdata);
+}
+
+/** @brief tensordec-plugin's TensorDecDef callback */
+static GstFlowReturn
+pose_decode (void **pdata, const GstTensorsConfig * config,
+    const GstTensorMemory * input, GstBuffer * outbuf)
+{
+  pose_data *data = *pdata;
+  const size_t size = data->width * data->height * 4;   /* RGBA */
+  GstMapInfo out_info;
+  GstMemory *out_mem;
+  GArray *results = NULL;
+  const GstTensorMemory *detections = NULL;
+  float *arr;
+  int index, i, j;
+
+  g_assert (outbuf);
+  /* Ensure we have outbuf properly allocated */
+  if (gst_buffer_get_size (outbuf) == 0) {
+    out_mem = gst_allocator_alloc (NULL, size, NULL);
+  } else {
+    if (gst_buffer_get_size (outbuf) < size) {
+      gst_buffer_set_size (outbuf, size);
+    }
+    out_mem = gst_buffer_get_all_memory (outbuf);
+  }
+  g_assert (gst_memory_map (out_mem, &out_info, GST_MAP_WRITE));
+  /** reset the buffer with alpha 0 / black */
+  memset (out_info.data, 0, size);
+
+  results = g_array_sized_new (FALSE, TRUE, sizeof (pose), POSE_SIZE);
+  detections = &input[0];
+  arr = detections->data;
+  for (index = 0; index < POSE_SIZE; index++) {
+    int maxX = 0;
+    int maxY = 0;
+    float max = 0.0;
+    pose p;
+    for (j = 0; j < data->i_height; j++) {
+      for (i = 0; i < data->i_width; i++) {
+        float cen = arr[i * POSE_SIZE + j * data->i_width * POSE_SIZE + index];
+        if (cen > max) {
+          max = cen;
+          maxX = i;
+          maxY = j;
+        }
+      }
+    }
+    p.valid = TRUE;
+    p.x = maxX;
+    p.y = maxY;
+    p.prob = max;
+    g_array_append_val (results, p);
+  }
+
+  draw (&out_info, data, results);
+  g_array_free (results, TRUE);
+  gst_memory_unmap (out_mem, &out_info);
+  if (gst_buffer_get_size (outbuf) == 0)
+    gst_buffer_append_memory (outbuf, out_mem);
+  return GST_FLOW_OK;
+}
+
+static gchar decoder_subplugin_pose_estimation[] = "pose_estimation";
+/** @brief Pose Estimation tensordec-plugin TensorDecDef instance */
+static TensorDecDef poseEstimation = {
+  .modename = decoder_subplugin_pose_estimation,.type = OUTPUT_VIDEO,.init =
```

Comment:
now struct for decoder-sub param is changed, need to fix.

Reply:
is this updated?

Reply:
I will update.